jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -2745,57 +2742,59 @@ ecma_builtin_array_prototype_object_reduce_right (ecma_value_t this_arg, /**< th
       ecma_number_t *num_p = ecma_alloc_number ();
       ecma_value_t accumulator = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
 
-      /* 6 */
-      uint32_t index = len - 1;
-      bool zero_reached = !len;
+      /* 6. */
+      int32_t index = (int32_t) len - 1;
```

Comment:
`int32_t` is not enough for case `len` is more than 2147483647 (that is allowed by specification). `int64_t` would be correct for all possible array indices.